pkg/server/server.go

```diff
@@ -498,12 +528,14 @@ func (s *Server) Run(ctx context.Context) error {
 		}
 
 		return nil
-	})
+	}); err != nil {
+		return err
+	}
 
 	// ========================================================================================================
 	// TODO: split apart everything after this line, into their own commands, optional launched in this process
```

Comment:
Can we remove this TODO unless we actually plan to do it?

Reply:
oh, this will be a follow-up. Part of this PR is to make that possible.